key.services.yml

```diff
@@ -2,7 +2,8 @@ services:
 
   key_manager:
     class: Drupal\key\KeyManager
+    arguments: ['@entity.manager', '@config.factory']
```

Comment:
That is the beauty of the service container. The service container handles this whether invoked by \Drupal::service or injected into another class. It also abstracts \Drupal away from the class so that theoretically someone could use it outside of the Drupal stack if they had some services for entity.manager and config.factory that behaved similarly.